langchain/vectorstores/elastic_vector_search.py

```diff
@@ -574,3 +574,118 @@ def knn_hybrid_search(
             source=source,
         )
         return dict(res)
+
+
+    def create_knn_index(self) -> None:
+         """
+         Creates an Elasticsearch index. The name of the index is specified during the instantiation of the class.
+         """
+         self.client.indices.create(
+             index=self.index_name,
+             mappings= self.mapping
+         )
+
+
+    def add_texts(self,
+               texts: Iterable[str],
+               metadatas: Optional[List[Dict[Any, Any]]] = None,
+               model_id: Optional[str] = None,
+               refresh_indices: bool = False,
+               **kwargs: Any
+               ) -> None:
+  #             ) -> List[str]:
```

Comment:
This should effectively override the parent method. So the signature should be:
```
 def add_texts(
        self,
        texts: Iterable[str],
        metadatas: Optional[List[dict]] = None,
        refresh_indices: bool = True,
        ids: Optional[List[str]] = None,
        **kwargs: Any,
    ) -> List[str]:
```

And return generated uuids from indexing if the user didn't provide IDs to index.

Reply:
@benwtrent Thanks for the review. 
I was also discussing with @hwchase17 removing `ElasticKnnSearch` as a subclass of `ElasticVectorSearch` and just making it 
```
class ElasticKnnSearch(VectorStore, ABC):
...
```

Since the kNN code doesn't really pull much from the parent class, it seems to just be adding complexity. This may also simplify when we add text_expansion search support

Reply:
If this overrides the parent or not, it would be really weird for users of the client to have two different methods, named the same thing, with the same goal, but doing different things.

Reply:
Totally agree and this was an oversight of mine originally. 
I'm going to fix the two methods up so they behave in the same fashion.